libs/xc/xenctrl.mli

```diff
@@ -240,6 +240,12 @@ val pages_to_mib : int64 -> int64
 val watchdog : handle -> int -> int32 -> int
   
 
+(* New MSR external functions *)
+val xc_get_msr_arch_caps : handle -> int64
+val xc_cpu_policy_get_system: handle -> int -> int64 array
```

Comment:
We talked about this offline - in general I believe the return values of these functions should be `string` - holding a byte sequence with no additional assumed structure; likewise I belive all `int64 array` values should be `string` as well.

Reply:
This is the recipe to copy a blob structure into an OCaml string:

```
CAMLlocal1(str);
str = caml_alloc_string(sizeof blob_t);
memcpy(String_val(str), blob, sizeof blob_t);
CAMLreturn(str);
```